src/bin/pg_autoctl/string_utils.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * src/bin/pg_autoctl/string_utils.c
+ *   Implementations of utility functions for string handling
+ *
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the PostgreSQL License.
+ *
+ */
+
+#include <errno.h>
+#include <inttypes.h>
+#include <stdlib.h>
+
+#include "snprintf.h"
+#include "string_utils.h"
+
+/*
+ * intToString converts an int to an IntString, which contains a decimal string
+ * representation of the integer.
+ */
+IntString
+intToString(int64_t number)
+{
+	IntString intString;
+
+	intString.intValue = number;
+
+	pg_snprintf(intString.strValue, INTSTRING_MAX_DIGITS, "%" PRId64, number);
+
+	return intString;
+}
+
+
+/*
+ * converts given string to 64 bit integer value.
+ * returns 0 upon failure and sets error flag
+ */
+int64_t
+stringToInt(const char *str, bool *error)
+{
+	char *endptr;
+	bool foundError = false;
+	int64_t number;
```

Comment:
This should be `long long int` and later it should be checked that number doesn't overflow/underflow int64 (same for function below). See this PR for my fix on the Citus versions:
https://github.com/citusdata/citus/pull/3602/files

Reply:
ok